src/SharedUtilities/models.jl

```diff
@@ -226,7 +226,7 @@ have different dimensions - require defining a new method.
 """
 function initialize_prognostic(
     model::AbstractModel{FT},
-    state::Union{ClimaCore.Fields.Field, Vector{FT}},
+    state::Union{ClimaCore.Fields.Field, ClimaCore.Fields.FieldVector, Vector{FT}},
```

Comment:
just curious - do we use this function with state of type FieldVector? ditto for initialize_auxiliary

Reply:
I don't think we use this combination of types, but we get a type error if I don't include `FieldVector` here. I think it's because it is a type option for the `AbstractLandModel` version of this function (the aqua test error goes away if I remove it in both places, but then I get the method error described below when running `LSM_single_column_tutorial"), but I'm not totally sure. @charleskawczynski do you have any thoughts?

Reply:
Which aqua test is failing? Ambiguities?

Reply:
Yeah, I can push a version showing the exact ambiguity message

Reply:
Actually I'll just copy it here. This is with the current version of the code, but with `ClimaCore.Fields.FieldVector` removed from the second parameter of `initialize_prognostic` and `initialize_auxiliary` in this file only.

`julia> include("test/aqua.jl")

[ Info: Precompiling ClimaLSM [7884a58f-fab6-4fd0-82bb-ecfedb2d8430]

method_ambiguity = (initialize_auxiliary(model::ClimaLSM.AbstractLandModel{FT}, coords::Union{ClimaCore.Fields.Field, ClimaCore.Fields.FieldVector}) where FT @ ClimaLSM ~/clima/ClimaLSM.jl/src/ClimaLSM.jl:92, initialize_auxiliary(model::AbstractModel{FT}, state::Union{Vector{FT}, ClimaCore.Fields.Field}) where FT @ ClimaLSM ~/clima/ClimaLSM.jl/src/SharedUtilities/models.jl:253)

method_ambiguity = (initialize_prognostic(model::ClimaLSM.AbstractLandModel{FT}, coords::Union{ClimaCore.Fields.Field, ClimaCore.Fields.FieldVector}) where FT @ ClimaLSM ~/clima/ClimaLSM.jl/src/ClimaLSM.jl:77, initialize_prognostic(model::AbstractModel{FT}, state::Union{Vector{FT}, ClimaCore.Fields.Field}) where FT @ ClimaLSM ~/clima/ClimaLSM.jl/src/SharedUtilities/models.jl:227)

Aqua tests (performance): Test Failed at /Users/jsloan/clima/ClimaLSM.jl/test/aqua.jl:22
  Expression: length(ambs) == 0
   Evaluated: 2 == 0`

Reply:
I think the issue is the union, you could probably match the two signatures and split the union, or just remove the argument types for both methods and let all of dispatch happen based on `model`, that might be simplest.

Reply:
I think it makes most sense to just dispatch off of the `model` type, so I'll do that. @kmdeck is this okay with you?

Reply:
yes! I think the pairing is 
LandModel -> FieldVector
AbstractModel -> Field or Vector?

Reply:
If I write the functions with these combinations of type annotations, I get method ambiguities for both `initialize_prognostic` and `initialize_auxiliary` (see above in this thread). Just to confirm, are you okay with the solution of not annotating the `state/coord` parameter and only annotating the `model` type as `AbstractLandModel` or `AbstractModel`?

Reply:
Is a compromise using 
Abstract Land Model + FieldVector
Abstract Model + no type annotation?

Just wondering, otherwise let's make it clear in the docs that in one case we expect something like a named tuple (FieldVector), and in the other, something like an array (Field or Vector) and then remove the type annotation on the state/coord argument...

Reply:
Ahh yes, that's a great compromise.

In the docstrings for the AbstractModel methods, I'll note the types we expect for the second argument